sdks/java/harness/src/test/java/org/apache/beam/fn/harness/state/OrderedListUserStateTest.java

```diff
@@ -0,0 +1,645 @@
+    assertArrayEquals(
+        asList(A1, A2, A3, A4).toArray(),
+        Iterables.toArray(stateBeforeFirstClearRange, TimestampedValue.class));
+    assertArrayEquals(
+        asList(A1, A3, A4).toArray(), Iterables.toArray(userState.read(), TimestampedValue.class));
+    userState.add(B2);
+    assertArrayEquals(
+        asList(A1, A2, A3, A4).toArray(),
+        Iterables.toArray(stateBeforeFirstClearRange, TimestampedValue.class));
+    assertArrayEquals(
+        asList(A1, B2, A3, A4).toArray(),
+        Iterables.toArray(userState.read(), TimestampedValue.class));
+
+    // add to an existing timestamp, clear, and then add
+    userState.add(B1);
+    userState.clearRange(Instant.ofEpochMilli(1), Instant.ofEpochMilli(2));
+    assertArrayEquals(
+        asList(A1, A2, A3, A4).toArray(),
+        Iterables.toArray(stateBeforeFirstClearRange, TimestampedValue.class));
+    assertArrayEquals(
+        asList(B2, A3, A4).toArray(), Iterables.toArray(userState.read(), TimestampedValue.class));
+    userState.add(B1);
+    assertArrayEquals(
+        asList(A1, A2, A3, A4).toArray(),
+        Iterables.toArray(stateBeforeFirstClearRange, TimestampedValue.class));
+    assertArrayEquals(
+        asList(B1, B2, A3, A4).toArray(),
+        Iterables.toArray(userState.read(), TimestampedValue.class));
+
+    // add a duplicated value, clear, and then add
+    userState.add(A3);
+    userState.clearRange(Instant.ofEpochMilli(3), Instant.ofEpochMilli(4));
+    assertArrayEquals(
+        asList(A1, A2, A3, A4).toArray(),
+        Iterables.toArray(stateBeforeFirstClearRange, TimestampedValue.class));
+    assertArrayEquals(
+        asList(B1, B2, A4).toArray(), Iterables.toArray(userState.read(), TimestampedValue.class));
+    userState.add(A3);
+    assertArrayEquals(
+        asList(A1, A2, A3, A4).toArray(),
+        Iterables.toArray(stateBeforeFirstClearRange, TimestampedValue.class));
+    assertArrayEquals(
+        asList(B1, B2, A3, A4).toArray(),
+        Iterables.toArray(userState.read(), TimestampedValue.class));
+  }
+
+  @Test
+  public void testAddAndClearRangeAfterClear() throws Exception {
+    FakeBeamFnStateClient fakeClient =
+        new FakeBeamFnStateClient(
+            timestampedValueCoder,
+            ImmutableMap.of(
+                createOrderedListStateKey("A", 1),
+                Collections.singletonList(A1),
+                createOrderedListStateKey("A", 3),
+                Collections.singletonList(A3),
+                createOrderedListStateKey("A", 4),
+                Collections.singletonList(A4)));
+
+    OrderedListUserState<String> userState =
+        new OrderedListUserState<>(
+            Caches.noop(),
+            fakeClient,
+            "instructionId",
+            createOrderedListStateKey("A"),
+            StringUtf8Coder.of());
+
+    userState.clear();
+    userState.clearRange(Instant.ofEpochMilli(0), Instant.ofEpochMilli(5));
```

Comment:
You shouldn't need this line after you change `clear()` to add a deleteRange for (int64min, int64max). Just calling clear should be enough to have `userState.read()` return empty, right?

Reply:
Added a new test to cover async() after clear().